libraries/ESP8266mDNS/src/LEAmDNS2Host_Structs.cpp

```diff
@@ -0,0 +1,3256 @@
+{
+    for (clsServiceTxt* pTxt : m_Txts)
+    {
+        delete pTxt;
+    }
+    m_Txts.clear();
+
+    return clearCache();
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::clearCache
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::clearCache(void)
+{
+    if (m_pcCache)
+    {
+        delete[] m_pcCache;
+        m_pcCache = 0;
+    }
+    return true;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::add
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::add(clsLEAMDNSHost::clsServiceTxt* p_pTxt)
+{
+    bool bResult = false;
+
+    if (p_pTxt)
+    {
+        m_Txts.push_back(p_pTxt);
+        bResult = true;
+    }
+    return ((clearCache()) &&
+            (bResult));
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::remove
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::remove(clsServiceTxt* p_pTxt)
+{
+    bool                            bResult = false;
+
+    clsServiceTxt::list::iterator   it(p_pTxt
+                                       ? std::find(m_Txts.begin(), m_Txts.end(), p_pTxt)
+                                       : m_Txts.end());
+    if (m_Txts.end() != it)
+    {
+        m_Txts.erase(it);
+        delete p_pTxt;
+
+        bResult = true;
+    }
+    return ((clearCache()) &&
+            (bResult));
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::count
+
+*/
+size_t clsLEAMDNSHost::clsServiceTxts::count(void) const
+{
+    size_t  stResult = m_Txts.size();
+    return stResult;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::removeTempTxts
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::removeTempTxts(void)
+{
+    bool            bResult = true;
+
+    // Delete content
+    clsServiceTxt::list tempTxts;
+    for (clsServiceTxt* pTxt : m_Txts)
+    {
+        if (pTxt->m_bTemp)
+        {
+            tempTxts.push_back(pTxt);
+            delete pTxt;
+        }
+    }
+    // Remove objects from list
+    for (clsServiceTxt* pTempTxt : tempTxts)
+    {
+        m_Txts.remove(pTempTxt);
+    }
+    return ((clearCache()) &&
+            (bResult));
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::find
+
+*/
+clsLEAMDNSHost::clsServiceTxt* clsLEAMDNSHost::clsServiceTxts::find(const char* p_pcKey)
+{
+    clsServiceTxt* pResult = 0;
+
+    for (clsServiceTxt* pTxt : m_Txts)
+    {
+        if ((p_pcKey) &&
+                (0 == strcmp(pTxt->m_pcKey, p_pcKey)))
+        {
+            pResult = pTxt;
+            break;
+        }
+    }
+    return pResult;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::find (const)
+
+*/
+const clsLEAMDNSHost::clsServiceTxt* clsLEAMDNSHost::clsServiceTxts::find(const char* p_pcKey) const
+{
+    const clsServiceTxt*   pResult = 0;
+
+    for (const clsServiceTxt* pTxt : m_Txts)
+    {
+        if ((p_pcKey) &&
+                (0 == strcmp(pTxt->m_pcKey, p_pcKey)))
+        {
+            pResult = pTxt;
+            break;
+        }
+    }
+    return pResult;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::find
+
+*/
+clsLEAMDNSHost::clsServiceTxt* clsLEAMDNSHost::clsServiceTxts::find(const clsServiceTxt* p_pTxt)
```

Comment:
Fair enough.  I was thinking that it would be hard to send in an invalid pointer, but it may be that some services disappear behind the scenes and the app may still call them.